distribution/lib/Standard/Base/0.0.0-dev/src/Enso_Cloud/Internal/Existing_Enso_Asset.enso

```diff
@@ -82,12 +84,17 @@ type Existing_Enso_Asset
         handle_not_found _ = if_not_found
         error_handlers = Map.from_vector [["resource_missing", handle_not_found]]
 
-        uri = Utils.cloud_root_uri+"path/resolve"
-        payload = JS_Object.from_pairs [["path", path]]
-
-        # TODO remove workaround - this should be a Get endpoint, not Post
-        response = Context.Output.with_enabled <|
-            Utils.http_request_as_json HTTP_Method.Post uri payload error_handlers=error_handlers
+        uri = ((URI.from Utils.cloud_root_uri) / "path/resolve") . add_query_argument "path" path
+        r = Utils.http_request_as_json HTTP_Method.Get uri error_handlers=error_handlers
+        ## Workaround to keep compatibility with old cloud API
+           TODO remove it after https://github.com/enso-org/cloud-v2/pull/1236 has been deployed
+        response = r.catch Enso_Cloud_Error error-> case error of
+            Enso_Cloud_Error.Unexpected_Service_Error status _ ->
+                if status.code != 404 then r else
+                    old_uri = (URI.from Utils.cloud_root_uri) / "path/resolve"
+                    old_payload = JS_Object.from_pairs [["path", path]]
+                    Context.Output.with_enabled <| Utils.http_request_as_json HTTP_Method.Post old_uri old_payload error_handlers=error_handlers
+            _ -> r
```

Comment:
Temporary fallback to make the migration smooth: if the GET endpoint responds with 404, we fall back to the POST endpoint.

Reply:
The commit d8460d0d063216ec2d53101c7d865da3199f2173 should be reverted after https://github.com/enso-org/cloud-v2/pull/1236 is deployed.